crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -118,13 +118,13 @@ pub fn derive_bundle(input: TokenStream) -> TokenStream {
     {
         if *is_bundle {
             field_type_infos.push(quote! {
-                type_info.extend(#field_type::type_info());
+                type_info.extend(<#field_type>::type_info());
             });
             field_get_components.push(quote! {
                 self.#field.get_components(&mut func);
             });
             field_from_components.push(quote! {
-                #field: #field_type::from_components(&mut func),
+                #field: <#field_type>::from_components(&mut func),
```

Comment:
Here's the other line that you can specify `Bundle` on.

```suggestion
                #field: <#field_type as #ecs_path::bundle::Bundle>::from_components(&mut func),
```